### PR TITLE
Join sms_usage and notifications in cloud formation

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Join sms usage dataset to notifications dataset

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/309

# Test instructions | Instructions pour tester la modification

- Test on dev, [see new dataset](https://ca-central-1.quicksight.aws.amazon.com/sn/data-sets/sms-usage-notifications/view)
- look at new dataset after staging merge

# Release Instructions | Instructions pour le déploiement

None.

# Reviewer checklist | Liste de vérification du réviseur

- [ ] This PR does not break existing functionality.
- [ ] This PR does not violate GCNotify's privacy policies.
- [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [ ] This PR does not significantly alter performance.
- [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: aws/quicksight/data_source.tf [hunk 1]
```diff
@@ -36,10 +36,10 @@ resource "aws_quicksight_data_source" "rds" {
 
 resource "aws_s3_object" "manifest_file" {
   bucket = var.s3_bucket_sms_usage_id
-  key    = "quicksight/s3-manifest-sms-usage.json"     # replace with desired object key
-  source = "./s3-manifests/s3-manifest-sms-usage.json" # replace with path to local file
+  key    = "quicksight/s3-manifest-sms-usage-${var.env}.json"     # replace with desired object key
+  source = "./s3-manifests/s3-manifest-sms-usage-${var.env}.json" # replace with path to local file
```
Comment (jimleroyer) [inline aws/quicksight/data_source.tf:40]: 😅

File: aws/quicksight/dataset_refresh_order.txt [hunk 2]
```diff
@@ -14,3 +16,5 @@ All times are UTC.
 
 0735 - login_events
 0740 - users
+
+0745 - sms_usage_notifications
```
Comment (jimleroyer) [inline aws/quicksight/dataset_refresh_order.txt:20]: Makes sense!

File: aws/quicksight/dataset_sms_usage_notifications.tf [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+          ],
+
+
+          LogicalTableMap = {
+
+            nj-notifications-smsusage = {
+              Alias = "nj-notifications-sms-usage",
+              DataTransforms = [
+                {
+                  ProjectOperation = {
```
Comment (jimleroyer) [inline aws/quicksight/dataset_sms_usage_notifications.tf:53]: From the TF doc I read, not defining a project operation would make a default one with all the fields. I wonder if CloudFormation is the same as well, as we could skip this operation?

File: aws/quicksight/dataset_sms_usage_notifications.tf
Comment (jimleroyer) [inline aws/quicksight/dataset_sms_usage_notifications.tf:1]: Good organization to put this into its own file for the actual join!

File: aws/quicksight/data_source.tf (hunk 1 above)
Comment (sastels) [reply aws/quicksight/data_source.tf:40]: yeah...

File: aws/quicksight/dataset_sms_usage_notifications.tf (hunk 3 above)
Comment (sastels) [reply aws/quicksight/dataset_sms_usage_notifications.tf:53]: I shall try that!

**Pull request merged**